lib/loggly/common.js

```diff
@@ -206,11 +206,12 @@ common.loggly = function () {
     if (timerFunction === null) {
       timerFunction = setInterval(function () {
         sendBulkLogs();
+        if (timerFunction && !arrMsg.length) {
+          clearInterval(timerFunction)
+          timerFunction = null;
+        }
       },5000);
-    } else if (timerFunction && !arrMsg.length) {
-        clearInterval(timerFunction);
-        timerFunction = null;
-      }
+    }
     arrMsg.push(requestBody);
     if (arrMsg.length === arrSize) {
       sendBulkLogs();
```

Comment:
At line #217, sendBulkLogs() function is called inside the if (arrMsg.length === arrSize) condition which basically checks if the arrMsg(an array in which logs get pushed) array's length is equal to arrSize which is 100 then send the all logs instantly without waiting.
And at line #208, sendBulkLogs() function is called inside the setInterval function to keep calling the same function in every 5 seconds.
This is the behaviour of the library since the bulk mode was implemented.